### DISCUSSION TITLE
Inputs get empty when component is re-rendered in production

### DISCUSSION THREAD

**Discussion by AliKHODR** [livewire/livewire#4291] [1. Help] [answered]
Hi, i'm having this weird issue when my component get re-rendered in production environment. The fields that has already been filled get empty when i check the checkbox that re-renders my component. 
Let's say the user fills the first two fields, then decides to check the checkbox to add an address. Well, then the first two fields get empty (show nothing). I only encounter this issue in production environment. In development, it woks fine.

![Capture d’écran 2021-11-27 à 18 39 02](https://user-images.githubusercontent.com/55912430/143692237-3a9b87ec-9aa5-47d0-886b-28ad335ae618.png)

 Livewire component : 
![Capture d’écran 2021-11-27 à 18 38 23](https://user-images.githubusercontent.com/55912430/143693238-bea2e209-60c3-4960-adb7-aa0c1e64ba34.png)

Blade file : 
![Capture d’écran 2021-11-27 à 18 31 55](https://user-images.githubusercontent.com/55912430/143693718-263d8599-aca9-4710-b93b-0e11248cbd75.png)

I have no idea why this happens. Help please

**Comment by joshhanley** [comment, marked as answer]
@AliKHODR any inputs that are inside Livewire need to be controlled by Livewire otherwise it will wipe out any data in the inputs. 

So you should put `wire:model` on `companyName` input for example.

Hope this helps!

**Reply by AliKHODR** [reply]
Thanks a lot @joshhanley, i will try your solution. But can you please tell me it only happens in production environment?

**Reply by joshhanley** [reply]
@AliKHODR honestly, I have no idea haha. I'd expect that to happen both locally and in production.

**Comment by antedrnasin** [comment]
Sorry for necro posting but I didn't wanna start a new thread but I have the same but opposite problem: 
I have one multiple select box and date range input.

When selecting values in the multiple, after every selected value the new value is being sent to server but when that happens, date input is empty. When i remove wire:model from date input than the value stays as it is.

For now I found a workaround that everytime multiple select is changed (on change event) the value in date input is preserved:

```js
@this.set('daterange', document.getElementById('daterange').value);
```
This is a really simple livewire form with only 2 inputs.

In the Component I have
```php
public $accounts = [];
    public $daterange;
```
 using livewire 2
 
 What am I doing wrong here? as this should be pretty straightforward
